pandas/core/dtypes/common.py

```diff
@@ -1646,39 +1647,44 @@ def is_extension_type(arr):
     False
     >>> is_extension_type(np.array([1, 2, 3]))
     False
-    >>>
+
+    Categoricals
     >>> cat = pd.Categorical([1, 2, 3])
-    >>>
     >>> is_extension_type(cat)
     True
     >>> is_extension_type(pd.Series(cat))
-    True
+
+    pandas' Sparse arrays
     >>> is_extension_type(pd.SparseArray([1, 2, 3]))
     True
     >>> is_extension_type(pd.SparseSeries([1, 2, 3]))
     True
-    >>>
     >>> from scipy.sparse import bsr_matrix
     >>> is_extension_type(bsr_matrix([1, 2, 3]))
     False
     >>> is_extension_type(pd.DatetimeIndex([1, 2, 3]))
     False
+
+    pandas' datetime with timezone
     >>> is_extension_type(pd.DatetimeIndex([1, 2, 3], tz="US/Eastern"))
     True
-    >>>
     >>> dtype = DatetimeTZDtype("ns", tz="US/Eastern")
     >>> s = pd.Series([], dtype=dtype)
     >>> is_extension_type(s)
     True
     """
-
-    if is_categorical(arr):
-        return True
-    elif is_sparse(arr):
-        return True
-    elif is_datetimetz(arr):
-        return True
-    return False
+    # XXX: we have many places where we call this with a `.dtype`,
+    # instead of a type. Think about supporting that too...
+    from pandas.core.extensions import ExtensionArray, ExtensionDtype
+    return (isinstance(arr, ExtensionArray) or
```

Comment:
this needs just to satisfy ``is_array_like`` (which is an iterable with a dtype)